x-pack/plugins/apm/common/connections.ts

```diff
@@ -25,7 +25,8 @@ export interface ServiceNode extends NodeBase {
 
 export interface BackendNode extends NodeBase {
   type: NodeType.backend;
-  backendName: string;
+  displayName: string;
+  resourceIdentifierFields: Record<string, string>;
```

Comment:
Resource identifiers might be serialised on the backend instead of the UI (also types should be restricted)